Gemfile

```diff
@@ -3,4 +3,6 @@ gemspec
 
 gem "jekyll-github-metadata", ">= 2.15"
 
+gem "jekyll-include-cache", group: :jekyll_plugins
+
```

Comment:
It seems like this gem causes the `github-pages` guild to fail (see the CI logs for this PR). I wonder if we can include a `if` guard to check if the variable exists, and if not, fall back to the non-cached version?

Reply:
Thanks for noticing! The CI error report is:

```
 Liquid Exception: Liquid syntax error (/home/runner/work/just-the-docs/just-the-docs/_includes/components/sidebar.html line 26): Unknown tag 'include_cached' included in /_layouts/default.html
```

I was getting that error when building with Jekyll 4 until I added `, group: :jekyll_plugins`. Strangely, the other CI builds with Jekyll 3.9.3 succeeded.

In any case, `github-pages` requires Jekyll 3, which we've [deprecated](https://just-the-docs.github.io/just-the-docs/MIGRATION/#deprecations). So `github-pages` is _already_ implicitly deprecated too. The sooner we drop support for it, the better!

If you think it's necessary to continue support for `github-pages`, perhaps one of the other maintainers could take a look and work out how to use it together with `jekyll-include-cache`?  I'm not a Ruby programmer, and I've never tried to understand how gems and Jekyll interact…

Reply:
Gotcha. I'm hesitant to drop support for `github-pages` now (I'd like to give the actions feature a bit more time to mature; to me, this feels like a `v1` release target for us). Once we merge in the previous PR, I can take a shot at conditionally caching if the plugin is loaded; I recall the ability to runtime-check what plugins have been loaded, though this may or may not be allowed by the `github-pages` gem itself.

Reply:
I now think the Jekyll error report is misleading, and the problem is the inconsistency of the following with the `github-pages` gem dependencies:

```ruby
gem "jekyll-github-metadata", ">= 2.15"
```

See my comments further down in the conversation for further details.

Reply:
I was mistaken (I had an outdated `GitHub-pages` gem).